contract/models/abstract_contract.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2004-2010 OpenERP SA
+# Copyright 2014 Angel Moya <angel.moya@domatix.com>
+# Copyright 2015-2020 Tecnativa - Pedro M. Baeza
+# Copyright 2016-2018 Carlos Dauden <carlos.dauden@tecnativa.com>
+# Copyright 2016-2017 LasLabs Inc.
+# Copyright 2018 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class ContractAbstractContract(models.AbstractModel):
+    _inherit = "contract.recurrency.basic.mixin"
+    _name = "contract.abstract.contract"
+    _description = "Abstract Recurring Contract"
+
+    # These fields will not be synced to the contract
+    NO_SYNC = ["name", "partner_id", "company_id"]
+
+    name = fields.Char(required=True)
+    # Needed for avoiding errors on several inherited behaviors
+    partner_id = fields.Many2one(
+        comodel_name="res.partner", string="Partner", index=True
+    )
+    pricelist_id = fields.Many2one(comodel_name="product.pricelist", string="Pricelist")
+    contract_type = fields.Selection(
+        selection=[("sale", "Customer"), ("purchase", "Supplier")],
+        default="sale",
+        index=True,
+    )
+
+    journal_id = fields.Many2one(
+        "account.journal",
+        string="Journal",
+        default=lambda s: s._default_journal(),
+        domain="[('type', '=', contract_type)," "('company_id', '=', company_id)]",
+        index=True,
+    )
+    company_id = fields.Many2one(
+        "res.company",
+        string="Company",
+        required=True,
+        default=lambda self: self.env.company.id,
+    )
+    line_recurrency = fields.Boolean(
```

Comment:
After the first invoice, only setting this option to true is possible.

The other case, we need to check that all lines have the same recurrence values (if not, ask the user to handle it manually before).

Reply:
They should have, as computed writable fields are for that. In fact, the contract lines to invoice code is not changed. We are only playing with assignations of these values from parent to children. Have you found any case where this propagation hasn't happened?

Reply:
What I mean, after the the first invoice, contract line may have different next invoice date and different next invoicing period. If the user un-check `line_recurrency` in the contract, the compute method will override the recurrence values.

Reply:
I think that we shouldn't avoid that situations, as if someone changes the line recurrence flag, it's their duty to check current state, and if a forbidden situation is got, then the constraint will arise.